javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -783,7 +824,51 @@ public boolean isTraversableAgain() {
 
     @Override
     public Iterator<T> iterator() {
-        return isEmpty() ? Iterator.empty() : trie.iterator();
+        return isEmpty() ? Iterator.empty()
+                         : new VectorIterator();
+    }
+
```

Comment:
Will go to sleep now and go on tomorrow - first 'working' day after 4 weeks vacation :) I'm mostly offline during the day. My mobile phone is broken (I wait until the release of iPhone 7), also I have no Internet (for private use) at $work.

Great work so far!!!